### DISCUSSION TITLE
Method/ability to share secrets across multiple repositories in an Organization

### DISCUSSION THREAD

**Discussion by rdmurphy** [community/community#26971] [Actions]
<p>Hello! This is one part question, but (I think) more a feature request.</p>
<p>I’ve had a lot of luck experimenting with GitHub Actions for a few things I’ve used other CI platforms for, but one thing I’ve run up against is (what I think) is the lack of an ability to centralize and share secrets. For example — I’d like to be able to set up credentials to push static assets to S3, but I’d like to be able to not have to potentially go to 20+ different repositories and add those secrets manually. Not only would that be tedious, it’d be difficult to swap them in a single swoop if necessary.</p>
<p>Perhaps it would be a feature that’d only make sense for Organizations (AKA every repository in an organization that meets X criteria will be able to tap into Y secrets) because that is a natural grouping.</p>
<p>The use case here isn’t just laziness. :wink: I have a scenario where a new repository can get programmatically created (like, say, with a CLI tool) by a user of the Organization, who will do some work that’d eventually get compiled and deployed as a static site. Ideally it wouldn’t be up to them to also then go put in the credentials every single time they create one.</p>
<p>But also happy to hear any suggestions of how to solve such a thing with the current offerings! Thank you!</p>

**Comment by mscoutermarsh** [comment]
<p>Organization wide secrets could definitely be useful. We’ve talked about this internally a few times.</p>
<p>Thanks for sharing your use case, makes our jobs easier. :heart:</p>
<p>I don’t have a timeline for this as at the moment. But can share it’s on the backlog for us to figure out.</p>

**Comment by RichiCoder1** [comment]
<p>+1. We have multiple repositories that consumer AWS and our internal package provider api keys and currently that would mean having our Our Cloud Engineering team manually copy secrets into multiple repositories. The ability to share secrets on an Org level and/or the ability to set secrets through an API would be super helpful.</p>

**Comment by SteveShaffer** [comment]
<p>Organization-wide secrets for GitHub Actions would be a hard requirement for my organization (and I imagine a large number of others) to consider moving our Jenkins CI pipelines to GitHub Actions.</p>

**Comment by ejhayes** [comment]
<p>+1 this would be very helpful. Having to recreate the same secrets (such as AWS access key/secret) across all of our repositories would definitely be a bit of a headache to maintain.</p>

**Comment by ffMathy** [comment]
<p>Any status update on this? AppVeyor already supports this.</p>

**Comment by mscoutermarsh** [comment]
<p>Thanks everyone. Yes, we’ll build this. No timeline to share yet.</p>

**Comment by ajcann** [comment]
<p>Even just exposing secret creation at the api level would at least allow a way to manage secrets across many repos. Our org has nearly 100 repos and I’m dreading trying to maintain common secrets across them - or worse worry over secret rotation in a sane way.</p>

**Comment by mscoutermarsh** [comment]
<p>Yes. We’re working on it. :+1:</p>

**Comment by ffMathy** [comment]
<p>Is there an ETA?</p>
<p>And by “Working on it”, do you mean it’s just in your backlog, or do you actually mean truly actively working on it?</p>

**Comment by ffMathy** [comment]
<p>Or maybe just letting us know how long it will take? 5 years? 2 years? 1 month? 3 months?</p>
<p>Then we can at least consider workarounds in the meanwhile, if it’s going to take too long.</p>

**Comment by hitmands** [comment]
<p>At my Company, <br>
we are deciding which vcs platform to use and the CI/CD plays a huge role in this.</p>
<p>Can you please share a date for the landing of this feature?</p>

**Comment by chiefjester** [comment]
<p>Hi, <a class="mention" href="/mscoutermarsh">@mscoutermarsh</a> Do you have anything to share now? Especially that you’ve released an API for secrets? While we can theoretically build it, we’re still leaning on a built-in solution by Github.</p>

**Comment by mscoutermarsh** [comment]
<p>Hi <a class="mention" href="/thisguychris">@thisguychris</a>,</p>
<p>Nothing yet. We did the first release of the API as repository only so that we’d get it shipped faster. We still plan to do a later update enabling secrets management for orgs as well.</p>
<p>We have a bunch of org level features on the roadmap. I don’t have a timeline for this yet, sorry.</p>

**Comment by mscoutermarsh** [comment]
<p>Sorry, I missed this notification.</p>
<p>It was on our backlog. Org level secrets are now actively being worked on.</p>

**Comment by alexrosenfeld10** [comment]
<p>Just adding my own +1 on this. It’s been a few months, is there a more narrow ETA on this? If not, no worries. Just an eager user here :slight_smile:</p>

**Comment by rodrigorodriguescosta** [comment]
<p>I saw there is that feature, that’s great!</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Organization level secrets are now available.</p>
<p>See here: <a href="https://github.blog/changelog/2020-05-14-organization-secrets/" rel="nofollow noopener">https://github.blog/changelog/2020-05-14-organization-secrets/</a></p>

**Comment by rdmurphy** [comment]
<p>Thank you! This is great. :tada:</p>
